site/source/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.rst

```diff
@@ -674,32 +674,43 @@ Here ``my_function`` is a C function that receives a single integer parameter
 integer. This could be something like ``int my_function(char *buf)``.
 
 The converse case of exporting allocated memory into JavaScript can be
-tricky when wasm-based memory is allowed to grow (by compiling with
-``-s ALLOW_MEMORY_GROWTH=1``). Increasing the size of memory changes
+tricky when wasm-based memory is allowed to **grow**, by compiling with
+``-s ALLOW_MEMORY_GROWTH``. Increasing the size of memory changes
 to a new buffer and existing array views essentially become invalid,
 so you cannot simply do this:
 
 .. code-block:: javascript
 
    function func() {
-     var ptr = callSomething(len);               // if memory grows ...
-     return HEAPU8.subarray(ptr, ptr+len); // ... this will fail
+     let someView = HEAPU8.subarray(x, y);
+     compute(someView);
+
+     // This may grow memory, which would invalidate all views.
+     maybeGrow();
+
+     // If we grew, this use of an invalidated view will fail.
```

Comment:
fail how? Throw because it's detached, right?

Reply:
I think there are multiple possible failures here. Some operations throw (like `.subarray`), but not typed array accesses - the array length gets switched to 0, and so its a typed array read or write out of bounds, which (sadly) does not trap but returns 0 for a read, or does nothing for a write.

I can add more detail here - what do you think?

Reply:
wow I forgot about that silent failure, that's terrible. (What is this, C?)
Yeah I think it would be worth adding a bit more detail on that.